### PR TITLE
suricata: adds quadfuzz

### PR DESCRIPTION
**Pull request by catenacyber**
One random build out of 10 adds a LLVM Custom Mutator to find quadratic complexities

### CODE REVIEW

File: projects/suricata/build.sh [hunk 1]
```diff
@@ -91,6 +91,14 @@ then
     fuzz_branches+=("6")
 fi
 
+if [[ `shuf -i 0-9 -n 1` -eq 0 ]]
```
Comment (catenacyber) [inline projects/suricata/build.sh:94]: is there a better way such as an environment variable ? cc @Alan32Liu

**Comment by catenacyber** [comment]
See also https://github.com/google/fuzzbench/pull/1854

File: projects/suricata/build.sh (hunk 1 above)
Comment (Alan32Liu) [reply projects/suricata/build.sh:94]: Maybe try `CFLAGS` and `CXXFLAGS`?
Here is [a similar example](https://github.com/google/oss-fuzz/blob/e8d9b9749abd171b4606def48b9648ec44f517af/infra/base-images/base-builder/compile_centipede#L27C95-L27C125).

Adding it to `LIB_FUZZING_ENGINE` is indeed too hacky.

**Comment by Alan32Liu** [comment]
@victorjulien, would this look good to you?
Thanks!

**Comment by victorjulien** [comment]
> @victorjulien, would this look good to you? Thanks!

Yes, @catenacyber has been running this with some good success.

**Comment by catenacyber** [comment]
Success for the fuzzer ;-)

**Comment by catenacyber** [comment]
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=63113 is nice

**Pull request merged**